phpunit.xml.dist

```diff
@@ -4,7 +4,9 @@
 	xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/6.3/phpunit.xsd"
 	backupGlobals="true"
 	beStrictAboutTestsThatDoNotTestAnything="false"
-	colors="true">
+	colors="true"
+	bootstrap="vendor/squizlabs/php_codesniffer/tests/bootstrap.php"
```

Comment:
Could you please elaborate on that -- phpunit seems to be always parsing the config file even if installed globally. That is how it is picking up [all the other config](https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/blob/17826a78d37a168b5b60f7a6cea7e9a8b0d545d8/phpunit.xml.dist) and would do with the `bootstrap` param too.

For example, the `WordPress` test suite is defined here:

https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/blob/17826a78d37a168b5b60f7a6cea7e9a8b0d545d8/phpunit.xml.dist#L10-L12

So `--filter WordPress` seems to apply only if the config file was parsed and used.

Similarly, we can move the `bootstrap` definition to `phpunit.xml.dist`.

Reply:
No, you can't as the bootstrap file is a file from the PHPCS repo and the `vendor` directory won't exist in non-Composer based setups, so you'd be pointing PHPUnit to a non-existing file which will prevent PHPUnit from running.

Reply:
> The particular run-tests composer script is currently already referencing the bootstrap file in ./vendor/squizlabs/php_codesniffer/tests/bootstrap.php so it would fail even now, if the file wasn't there.

You can safely presume a composer script will only be used by devs who are using a Composer-based install, so it's fine to reference it in the composer `run-tests` script.

The `phpunit.xml.dist` file, however, is used by devs with any type of setup, including non-Composer based installs.
Referencing a file in the `vendor` directory in the `phpunit.xml.dist` file would *force* every dev using a non-Composer based install to have custom a `phpunit.xml` file to force overrule the bootstrap directive.

Reply:
> Referencing a file in the vendor directory in the phpunit.xml.dist file would force every dev using a non-Composer based install to have custom a phpunit.xml file to force overrule the bootstrap directive.

Don't they already have to provide a bootstrap file either as a command line param or in a custom `phpunit.xml` since it isn't specified anywhere by default.

Reply:
They probably do, but your change means they can no longer rely on the default `phpunit.xml.dist` file , which means they will most likely need to make changes in their setup.

Your basically mixing in a Composer-setup-specific change in the generic dev environment which is also used by non-Composer setups. That just complicates things unnecessarily.